### PR TITLE
Fix pandas Series materializer

### PR DESCRIPTION
**Pull request by jordandelbar**
## Describe changes
Fix the #1145 issue

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by schustmi** [comment]
Thanks for the PR @jordandelbar! It would be great if you could rebase your changes on the latest develop state and add a quick test for this change in `tests/unit/materializers/test_pandas_materializer.py`. We have an utility function to quickly test materializers (check [this](https://github.com/zenml-io/zenml/blob/develop/tests/unit/materializers/test_built_in_materializer.py) out for some inspiration on how to use it) that should hopefully make this very easy

**Review by safoinme — APPROVED** [review]
LGTM! Thank you for fixing this 🙌

**Comment by jordandelbar** [comment]
The bug only appears when `pyarrow` is also installed in the environment.
The unit test I added must be run in a `venv` with `pyarrow` and one without `pyarrow` (`pyarrow` that could come from another dependency). 
It could be done by adding a dimension in your matrix strategy of your `.github/workflows/unit-test.yml`

File: tests/unit/materializers/test_pandas_materializer.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#  permissions and limitations under the License.
+
+import pandas
+
+from tests.unit.test_general import _test_materializer
+from zenml.materializers.pandas_materializer import PandasMaterializer
+
+
+def test_pandas_materializer():
+
```
Comment (safoinme) [inline tests/unit/materializers/test_pandas_materializer.py:22]:
```suggestion
"""Test the pandas materializer."""
```
Adding the docstring to the test function, can you please also try running the `scripts/format.sh` so the code can be in the same format

**Review by safoinme — CHANGES_REQUESTED** [review]

File: tests/unit/materializers/test_pandas_materializer.py (hunk 1 above)
Comment (jordandelbar) [reply tests/unit/materializers/test_pandas_materializer.py:23]: My bad, it's done

**Review by safoinme — APPROVED** [review]

**Comment by schustmi** [comment]
We're currently developing a new integration testing framework which can also be used to run specific unit tests in special environments, this might be a candidate for that to check that the materializer works with and without `pyarrow`.

**Pull request merged**